spec/unit/file_system_spec.rb

```diff
@@ -1065,12 +1065,60 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
           end
         end
 
-        it 'applies the specified mode' do
+        it 'applies 0644 mode' do
           Puppet::FileSystem.replace_file(dest, 0644) { |f| f.write(content) }
 
           expects_public_file(dest)
         end
 
+        it 'applies 0660 mode' do
+          Puppet::FileSystem.replace_file(dest, 0660) { |f| f.write(content) }
+          current_sid = Puppet::Util::Windows::SID.name_to_sid(Puppet::Util::Windows::ADSI::User.current_user_name)
+          sd = Puppet::Util::Windows::Security.get_security_descriptor(dest)
+
+          expect(sd.dacl).to contain_exactly(
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::LocalSystem, mask: 0x1f01ff),
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::BuiltinAdministrators, mask: 0x1f01ff),
+            an_object_having_attributes(sid: current_sid, mask: 0x1f01ff),
+          )
+        end
+
+        it 'applies 0640 mode' do
+          Puppet::FileSystem.replace_file(dest, 0640) { |f| f.write(content) }
+          current_sid = Puppet::Util::Windows::SID.name_to_sid(Puppet::Util::Windows::ADSI::User.current_user_name)
+          sd = Puppet::Util::Windows::Security.get_security_descriptor(dest)
+
+          expect(sd.dacl).to contain_exactly(
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::LocalSystem, mask: 0x1f01ff),
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::BuiltinAdministrators, mask: 0x1f01ff),
+            an_object_having_attributes(sid: current_sid, mask: 0x1f01ff),
+          )
+        end
+
+        it 'applies 0600 mode' do
+          Puppet::FileSystem.replace_file(dest, 0600) { |f| f.write(content) }
+          current_sid = Puppet::Util::Windows::SID.name_to_sid(Puppet::Util::Windows::ADSI::User.current_user_name)
+          sd = Puppet::Util::Windows::Security.get_security_descriptor(dest)
+
+          expect(sd.dacl).to contain_exactly(
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::LocalSystem, mask: 0x1f01ff),
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::BuiltinAdministrators, mask: 0x1f01ff),
+            an_object_having_attributes(sid: current_sid, mask: 0x1f01ff),
+          )
+        end
+
+        it 'applies 0440 mode' do
+          Puppet::FileSystem.replace_file(dest, 0440) { |f| f.write(content) }
+          current_sid = Puppet::Util::Windows::SID.name_to_sid(Puppet::Util::Windows::ADSI::User.current_user_name)
+          sd = Puppet::Util::Windows::Security.get_security_descriptor(dest)
+
+          expect(sd.dacl).to contain_exactly(
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::LocalSystem, mask: 0x1f01ff),
+            an_object_having_attributes(sid: Puppet::Util::Windows::SID::BuiltinAdministrators, mask: 0x1f01ff),
+            an_object_having_attributes(sid: current_sid, mask:  0x1f01ff)
+          )
+        end
```

Comment:
I could DRY this after Jenkins runs, using a share example or an array of modes.